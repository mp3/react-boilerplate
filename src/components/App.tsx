import Top from 'components/pages/Top'
import * as React from 'react'
import { hot } from 'react-hot-loader'
import { Route, Switch } from 'react-router'

export default hot(module)(() => (
  <Switch>
    <Route exact={true} path="/" component={Top} />
  </Switch>
))
