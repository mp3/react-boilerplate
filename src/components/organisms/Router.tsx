import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import createStore from 'store/createStore'

const history = createBrowserHistory()
export const store = createStore(history)

interface Props {
  children: JSX.Element
}

export default (props: Props) => {
  const { children } = props
  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>{children}</ConnectedRouter>
    </Provider>
  )
}
