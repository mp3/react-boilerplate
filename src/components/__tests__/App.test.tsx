import * as React from 'react'
import * as renderer from 'react-test-renderer'
import App from 'components/App'
import Router from 'components/organisms/Router'

test('render <App />', () => {
  const tree = renderer.create(
    <Router>
      <App />
    </Router>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
