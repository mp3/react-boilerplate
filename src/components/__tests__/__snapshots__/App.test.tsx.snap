// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <App /> 1`] = `null`;
